draft-irtf-cfrg-vdaf.md

```diff
@@ -1190,6 +1194,190 @@ how the nonces are chosen, but security requires that the nonces be unique. See
+  round and/or the prep message for the current round; correspondingly, each
+  response consists of the prep message for the current round and the Helper's
+  prep share for the next round.
+
+The Aggregators proceed in this ping-ponging fashion until a step of the
+computation fails, indicating the report is invalid and should be rejected, or
+the last round is reached and each Aggregator computes an output share. All
+told there there are `ceil((Vdaf.ROUNDS+1)/2)` requests sent.
+
+~~~ transition
+State CONTINUED(round_num, host_prep_state)
+
+def ping_pong_prep_next(Vdaf,
+                        is_leader: bool,
+                        agg_param: Vdaf.AggParam,
+                        inbound_type: Optional[str],
+                        inbound: Optional[bytes]):
+    if inbound_type == "initialize":
+        prep_msg, peer_prep_share = None, inbound
+    elif inbound_type == "continue":
+        l = Vdaf.prep_size(agg_param, round_num)
+        prep_msg, prep_share = inbound[:l], inbound[l:]
+        round_num += 1
+    elif inbound_type == "finish":
+        prep_msg, peer_prep_share = inbound, None
+    else:
+        prep_msg, peer_prep_share = None, None
+
+    try:
+        # If prep_msg == None then this is the start of the
+        # first round. Otherwise, this is the prep message
+        # from the end of the previous round.
+        out = Vdaf.prep_next(host_prep_state, prep_msg)
+        if round_num == Vdaf.ROUNDS:
+            state FINISHED(out)
+            # The Leader reaches this state upon processing a
+            # response from the Helper, so there is no outbound
+            # message.
+            return (None, None) if is_leader else ("finished", None)
+        (host_prep_state, host_prep_share) = out
+        round_num += 1
+
+        if peer_prep_share != None:
+            prep_shares = [peer_prep_share, host_prep_share]
+            if is_leader:
+                prep_shares.reverse()
+            prep_msg = Vdaf.prep_shares_to_prep(
+                agg_param,
+                prep_shares,
+            )
+            out = Vdaf.prep_next(host_prep_state, prep_msg)
+            if round_num == Vdaf.ROUNDS:
+                state FINISHED(out)
+                return ("finish", prep_msg)
+            (host_prep_state, host_prep_share) = out
+
+        state CONTINUED(round_num, host_prep_state)
+        if is_leader and round_num == 1:
+            return ("initialize", host_prep_share)
+        else:
+            return ("continue", host_prep_share)
+    except:
+        state REJECTED
+        return (None, None)
+~~~
+
+To initialize preparation, the Leader would call
+
+~~~
+(outbound_type, outbound) = Vdaf.ping_pong_prep_next(True,
+                                                     agg_param,
+                                                     None,
+                                                     None)
+~~~
+
+in state `CONTINUED(0, leader_prep_state)` and send `outbound` to the Helper.
```

Comment:
This should probably be reworded, "to initialize preparation" seems to be counter to already being in the state CONTINUED. If anything is initializing preparation, it seems like it ought to be `prep_start`. Maybe, "At the start of a request/response cycle, ..."?

Reply:
Done. Changed to "At the start of the **first** request/response cycle..."